lib/tasks/archive_submissions.rake

```diff
@@ -0,0 +1,82 @@
+# Copy all collected submissions from this MarkUs instance to the +target_dir+
+# organized by assignment and grouping. If a relative_url_root is set for this instance
+# the files will be copied into +target_dir+/relative_url_root, otherwise they will be
+# put directly into the +target_dir+.
+#
+# Group names are anonymized by naming each group directory as "group" plus integers in
+# ascending order. A file named group_memberships.csv will be written to the top level
+# directory which will contain a mapping of group directory names to student usernames.
+# For example:
+#
+# group0,student1,student2
+# group1,student3,
+#
+# Example usage:
+#   (relative_url_root is "/markus_instance" for this example)
+#
+#   $ bundle exec rails markus:archive_submissions[/home/userA/markus_archive]
+#
+#   This will create the following tree structure:
+#
+#   /home/userA/markus_archive/markus_instance
+#   └── A0
+#       ├── group0
+#       │   ├── file1.py
+#       │   └── some_inner_dir
+#       │       ├── file2.py
+#       │       └── file3.py
+#       ├── group1
+#       │   ├── file1.py
+#       │   └── some_inner_dir
+#       │       ├── file2.py
+#       │       └── file3.py
+#       └── group2
+#           ├── file1.py
+#           └── some_inner_dir
+#               ├── file2.py
+#               └── file3.py
+
+namespace :markus do
+  task :archive_submissions, [:target_dir] => :environment  do |_task, args|
+    target_dir = Pathname.new(args[:target_dir])
+    raise "Directory #{target_dir} does not exist" unless Dir.exist?(target_dir)
+
+    relative_root = Rails.application.config.action_controller.relative_url_root
+    if relative_root
+      target_dir += relative_root
+      FileUtils.makedirs target_dir
+    end
+
+    total_groupings = Grouping.joins(current_submission_used: :submission_files).uniq.count
```

Comment:
My understanding here is that the intention with the `:submission_files` is to ignore groups who didn't submit anything. That's good, but in this case there should be another check in the `Grouping` loop below that does something like `next unless submission.submission_files`, otherwise the count might be off.